### DISCUSSION TITLE
[DISCUSS] Refactor the way we're building/using Thrift

### DISCUSSION THREAD

**Discussion by chrisdutz** [apache/plc4x#1046] [General]
Hi, 

so I've been fighting to simplify our use of Thrift, as there were several problems with how we do it today on my M2 Mac (Mostly however to an updated Homebrew, which uses different paths).

While going through the project, I noticed that everything in "protocol" downloads a pre-built executable, which is simply downloaded and run. 

I would like to move the compile-tools directory into the root of the project, but detach it from the rest of the project. This only needs to be run every now and then, if something fundamental changes (Like new OSes or new CPU Architectures).

I would like to update that module, so it will not only build an executable, but also package the includes and the libraries, so we can use that in the client-cpp and the client-cpp-example.

It does seem as if an x86_64 binary of thrift also works on aarm64, I'd like to give it a try, if we could also use this approach for the libs and completely get rid of the thrift compilation as part of the build.

What do you think?

Chris

**Comment by chrisdutz** [comment]
Argh ... wrong project ;-)

**Comment by ottobackwards** [comment]
I was very confused for a second
